services/provider/server/server.go

```diff
@@ -1813,6 +1826,42 @@ func (s *OCSProviderServer) appendVolumeReplicationClassKubeResources(
 	return kubeResources, nil
 }
 
+func (s *OCSProviderServer) appendVolumeGroupReplicationClassKubeResources(
+	ctx context.Context,
+	kubeResources []client.Object,
+	consumer *ocsv1alpha1.StorageConsumer,
+	consumerConfig util.StorageConsumerResources,
+	storageCluster *ocsv1.StorageCluster,
+	rbdStorageId string,
+	remoteRbdStorageId string,
+) ([]client.Object, error) {
+	if mirrorEnabled, err := s.isConsumerMirrorEnabled(ctx, consumer); err != nil {
+		return nil, err
+	} else if !mirrorEnabled {
+		klog.Infof("skipping distribution of VolumeGroupReplicationClass as mirroring is not enabled for the consumer")
+		return kubeResources, nil
+	}
```

Comment:
Why does the server condition the sending of the information to the client if mirroring is not enabled? I would say that this is more a confusing behavior then anything else. If the admin configured something to be sent it should be sent, if he didn't first enabled mirroring then it is his mistake but the system should continue to be consistent.